egs/uw3/v1/run.sh

```diff
@@ -13,119 +10,84 @@ exp_dir=exp
 
 if [ $stage -le 0 ]; then
   # Data preparation
-  local/prepare_data.sh --dir $data_download
+  local/prepare_data.sh --dir data --download-dir data/download
 fi
 
-mkdir -p $data_dir/{train,test}/data
+mkdir -p data/{train,test}/data
 if [ $stage -le 1 ]; then
   for f in train test; do
-    local/make_feature_vect.py --scale-size 40 --color 1 --pad true $data_download/$f | \
+    local/make_features.py --scale-size 40 --color 1 --pad true data/$f | \
       copy-feats --compress=true --compression-method=7 \
-      ark:- ark,scp:$data_dir/$f/data/images.ark,$data_dir/$f/feats.scp || exit 1
+      ark:- ark,scp:data/$f/data/images.ark,data/$f/feats.scp || exit 1
 
-    steps/compute_cmvn_stats.sh $data_dir/$f || exit 1;
+    steps/compute_cmvn_stats.sh data/$f || exit 1;
   done
 fi
 
-numLeavesTri=500
-numGaussTri=20000
-numLeavesMLLT=500
-numGaussMLLT=20000
-
 beam=50
 
 if [ $stage -le 2 ]; then
-  local/prepare_dict.sh $data_dir/train/ $data_dir/test/ $data_dir/train/dict
+  local/prepare_dict.sh data/train/ data/test/ data/train/dict
   utils/prepare_lang.sh --num-sil-states 4 --num-nonsil-states 8 --position-dependent-phones false \
-    $data_dir/train/dict "<sil>" $data_dir/lang/temp $data_dir/lang
+    data/train/dict "<sil>" data/lang/temp data/lang
 fi
 
 if [ $stage -le 3 ]; then
-  cp -R $data_dir/lang -T $data_dir/lang_test
+  mkdir -p data/lang_test
+  cp -R data/lang/. data/lang_test/
 
-  cp $data_dir/train/text $data_dir/train/text_copy
-  cat $data_dir/test/text | awk '{ for(i=2;i<=NF;i++) print $i;}' | sort -u >test_words.txt
-  cat $data_dir/train/text | awk '{ for(i=2;i<=NF;i++) print $i;}' | sort -u >train_words.txt
+  cp data/train/text data/train/text_copy
+  cat data/test/text | awk '{ for(i=2;i<=NF;i++) print $i;}' | sort -u >test_words.txt
+  cat data/train/text | awk '{ for(i=2;i<=NF;i++) print $i;}' | sort -u >train_words.txt
   filter_scp.pl --exclude train_words.txt test_words.txt >diff.txt
-  cat diff.txt | awk '{ print "id " $1 }' >> $data_dir/train/text_copy
+  cat diff.txt | awk '{ print "id " $1 }' >> data/train/text_copy
 
-  local/prepare_lm.sh $data_dir/train/text_copy $data_dir/lang_test || exit 1;
+  local/prepare_lm.sh data/train/text_copy data/lang_test || exit 1;
```

Comment:
Could you please run this with `data/train/text` instead of `text_copy` and share
the results? If the results are the same, we wouldn't need all the hassle to generate `text_copy`.

Reply:
The current system is closed vocabulary and adds the words in the test data. I remember that the WER improved when I made that change though I can't remember by how much. I'll run the scripts again and let you know.